engine/action_contact.go

```diff
@@ -403,6 +403,48 @@ func (c *Client) upsertContact(ctx context.Context, paymailService paymailclient
 	return contact, nil
 }
 
+// AdminConfirmContacts confirms provided contacts, should be used only by the admin.
+func (c *Client) AdminConfirmContacts(ctx context.Context, contacts []*Contact) error {
+	contactList := make([]*Contact, 0, len(contacts))
+	for _, contact := range contacts {
+		cont, err := getContact(ctx, contact.Paymail, contact.OwnerXpubID, c.DefaultModelOptions()...)
+		if err != nil {
+			c.logContactError(contact.OwnerXpubID, contact.Paymail, fmt.Sprintf("error while getting contact: %s", err.Error()))
+			return spverrors.Wrapf(err, "error while getting contact")
+		}
+
+		if cont == nil {
+			return spverrors.ErrContactsNotFound
+		}
+
+		contactList = append(contactList, cont)
+	}
```

Comment:
If I understand correctly, this approach would result in N queries to the database, where N is the number of contacts. If that's the case, I’d suggest considering a bulk query to retrieve the data with the minimal number of calls. Do we currently support this kind of functionality in our database implementation? 🤔

Reply:
I wouldn't do this right now - because we want to make a transition from custom ("buggy") `dataStore` package to a approach leveraging GORM.
So, for now, I would suggest not to touch the `datasStore` package specially I wouldn't add any new features to it.

Reply:
Since there is currently a limit of 2 contacts to confirm in a request, the current implementation is not as much of a pain. As Chris mentioned, adding features to the `dataStore` is a waste of time.